processing/src/main/java/io/druid/query/filter/NotDimFilter.java

```diff
@@ -69,6 +79,24 @@ public Filter toFilter()
   }
 
   @Override
+  public RangeSet<String> getDimensionRangeSet(String dimension)
+  {
+    if (field instanceof AndDimFilter) {
```

Comment:
Why is the special casing for And/Or/Not needed rather than always taking the complement of the field's rangeSet?